### PR TITLE
Fixed health_lost, added a few more events

### PR DESCRIPTION
**Pull request by edunad**
Added randint, allows you to specify on random a min and a max
Added random_timer event - A timer that triggers between a min and a max for x time.

randomtimer

ballsssssssssssssssssssssss

Fixed health_lost , randomly worked sometimes, the event was getting triggered too fast, din't let parents trigger. Had to add a manual cooldown, works now.

Added owner_alive - Checks if the owner is alive or not

### CODE REVIEW

File: lua/pac3/core/client/parts/event.lua [hunk 1]
```diff
@@ -75,28 +123,46 @@ PART.Events =
 		end,
 	},
 
+	// Extreme weird hacky way to force the parents of the event to trigger.
```
Comment (edunad) [inline lua/pac3/core/client/parts/event.lua]: Attempted everything, this seems to be the only way to always be consistent

**Comment by suchipi** [comment]
Why do you do `if self.SetRandom == nil` and `if !self.SetRandom` ? Aren't these redundant? It also occurs with `if self.pac_wasdmg == nil` and `if !self.pac_wasdmg`.

**Comment by edunad** [comment]
mybad

**Comment by PotcFdk** [comment]
Can we take out the garrysyntax?

**Comment by suchipi** [comment]
I agree that those gravgun hooks don't belong there, but let's not do more than one thing with a PR, that just leads to a mess and will make git blaming those harder later.

File: lua/pac3/core/client/parts/event.lua (hunk 1 above)
Comment (suchipi) [inline lua/pac3/core/client/parts/event.lua]: Is this comment referring to what's above it or below?

Comment (edunad) [inline lua/pac3/core/client/parts/event.lua]: Below

**Comment by PotcFdk** [comment]
What? No, you have it backwards.
Use standard lua comments (`--` for single line, `--[[ ]]` for multi-line) instead of C-style ones.

File: lua/pac3/core/client/parts/event.lua (hunk 1 above)
Comment (Noiwex) [inline lua/pac3/core/client/parts/event.lua]: Oh for god's sake, when will you learn not to use piece of shit Garry's
syntax?

**Comment by suchipi** [comment]
Can you please not even put your comments there? The pull request should be topical in nature. If you have comments there, then that can mess up later rebases, because the lines are different... just please only modify what you actually need to modify and make a separate pull request for each topic.

**Comment by edunad** [comment]
Sorry, im used to comment everything i do, so i wont forget it later. Which is a good thing actually, but ok

**Comment by suchipi** [comment]
Sorry, I should have clarified. I was only referring to your comments about the gravgun hooks. Your other comments are, as you say, a good thing.

**Comment by Python1320** [comment]
I'll merge it from upstream.

**Pull request closed without merging**